File: examples/quickstart.py
"""TreeDex Quick Start Example.

Usage:
    pip install treedex
    export GEMINI_API_KEY="your-key"
    python examples/quickstart.py path/to/document.pdf
"""

import os
import sys

from treedex import TreeDex, GeminiLLM


def main():
    if len(sys.argv) < 2:
        print("Usage: python quickstart.py <path-to-pdf>")
        sys.exit(1)

    pdf_path = sys.argv[1]
    api_key = os.environ.get("GEMINI_API_KEY")
    if not api_key:
        print("Set GEMINI_API_KEY environment variable")
        sys.exit(1)

    # 1. Create LLM backend
    llm = GeminiLLM(api_key=api_key)

    # 2. Build index
    print(f"Indexing: {pdf_path}")
    index = TreeDex.from_file(pdf_path, llm=llm)

    # 3. Show structure
    print("\n--- Document Structure ---")
    index.show_tree()

    # 4. Show stats
    print("\n--- Stats ---")
    for k, v in index.stats().items():
        print(f"  {k}: {v}")

    # 5. Interactive query loop
    print("\n--- Query Mode (type 'quit' to exit) ---")
    while True:
        question = input("\nQuestion: ").strip()
        if question.lower() in ("quit", "exit", "q"):
            break

        result = index.query(question)
        print(f"\nSource: {result.pages_str}")
        print(f"Reasoning: {result.reasoning}")
        print(f"\nContext:\n{result.context[:1000]}")


if __name__ == "__main__":
    main()

File: examples/node/quickstart.ts
/**
 * TreeDex Quick Start Example (Node.js)
 *
 * Usage:
 *   npx tsx examples/node/quickstart.ts path/to/document.pdf
 *
 * Requirements:
 *   npm install treedex @google/generative-ai
 *   export GEMINI_API_KEY="your-key"
 */

import { TreeDex, GeminiLLM } from "../../src/index.js";
import { createInterface } from "node:readline";

async function main() {
  const pdfPath = process.argv[2];
  if (!pdfPath) {
    console.log("Usage: npx tsx examples/node/quickstart.ts <path-to-pdf>");
    process.exit(1);
  }

  const apiKey = process.env.GEMINI_API_KEY;
  if (!apiKey) {
    console.log("Set GEMINI_API_KEY environment variable");
    process.exit(1);
  }

  // 1. Create LLM backend
  const llm = new GeminiLLM(apiKey);

  // 2. Build index
  console.log(`Indexing: ${pdfPath}`);
  const index = await TreeDex.fromFile(pdfPath, llm);

  // 3. Show structure
  console.log("\n--- Document Structure ---");
  index.showTree();

  // 4. Show stats
  console.log("\n--- Stats ---");
  const stats = index.stats();
  for (const [k, v] of Object.entries(stats)) {
    console.log(`  ${k}: ${v}`);
  }

  // 5. Interactive query loop
  console.log("\n--- Query Mode (type 'quit' to exit) ---");
  const rl = createInterface({ input: process.stdin, output: process.stdout });

  const ask = (): void => {
    rl.question("\nQuestion: ", async (question) => {
      const q = question.trim();
      if (["quit", "exit", "q"].includes(q.toLowerCase())) {
        rl.close();
        return;
      }

      const result = await index.query(q);
      console.log(`\nSource: ${result.pagesStr}`);
      console.log(`Reasoning: ${result.reasoning}`);
      console.log(`\nContext:\n${result.context.slice(0, 1000)}`);
      ask();
    });
  };

  ask();
}

main().catch(console.error);
